app/models/job_invocation_composer.rb

```diff
@@ -221,6 +221,10 @@ def initialize(feature_label, hosts, provided_inputs = {})
         @host_bookmark = hosts
       elsif hosts.is_a? Host::Base
         @host_objects = [hosts]
+      elsif hosts.is_a? Array
+        @host_objects = hosts.map do |id|
+          Host::Managed.authorized.friendly.find(id)
```

Comment:
This is the key part - it should be able to find arrays of host names/ids. I switched from using the resource finder because it is only available in the controller.